clic/src/tier1/multiply_image_and_position.cpp

```diff
@@ -14,7 +14,21 @@ multiply_image_and_position_func(const Device::Pointer & device,
                                  Array::Pointer          dst,
                                  int                     dimension) -> Array::Pointer
 {
-  tier0::create_like(src, dst);
+  auto type = src->dtype();
+  switch (src->dtype())
+  {
+    case dType::INT8:
+    case dType::INT16:
+      type = dType::INT32;
+      break;
+    case dType::UINT8:
+    case dType::UINT16:
+      type = dType::UINT32;
+      break;
+    default:
+      break;
+  }
+  tier0::create_like(src, dst, type);
```

Comment:
I think we should keep it simple and aim for only one return type. Here the logic would point to UINT32 because we should return coordinates. The alternative could be FLOAT to allow negative and floating points.

```suggestion
  tier0::create_like(src, dst, dType::UINT32);
```

Reply:
You are right, one return type is much simpler and cleaner. I would suggest FLOAT here, otherwise we will make the results of center_of_mass() inaccurate.

I have made the changes accordingly in  my last commit